

<!DOCTYPE html>
<html class="writer-html5" lang="en" data-content_root="../../">
<head>
  <meta charset="utf-8" /><meta name="viewport" content="width=device-width, initial-scale=1" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Alignment Plots &mdash; ViennaRNA Package</title>
      <link rel="stylesheet" type="text/css" href="../../_static/pygments.css@v=80d5e7a1.css" />
      <link rel="stylesheet" type="text/css" href="../../_static/css/theme.css@v=e59714d7.css" />
      <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css@v=76b2166b.css" />
      <link rel="stylesheet" type="text/css" href="../../_static/css/custom.css@v=c1a09ec4.css" />

  
    <link rel="shortcut icon" href="../../_static/vrna_32.png"/>
      <script src="../../_static/jquery.js@v=5d32c60e"></script>
      <script src="../../_static/_sphinx_javascript_frameworks_compat.js@v=2cd50e6c"></script>
      <script src="../../_static/documentation_options.js@v=dfec817d"></script>
      <script src="../../_static/doctools.js@v=888ff710"></script>
      <script src="../../_static/sphinx_highlight.js@v=dc90522c"></script>
      <script src="../../_static/clipboard.min.js@v=a7894cd8"></script>
      <script src="../../_static/copybutton.js@v=f281be69"></script>
    <script src="../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Deprecated Interface for Plotting Utilities" href="deprecated.html" />
    <link rel="prev" title="Pair Probability Plots" href="probabilities.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../index.html" class="icon icon-home">
            ViennaRNA
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
    
              <p class="caption" role="heading"><span class="caption-text">Installation</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../install.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../configuration.html">Configuration</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Usage and Tutorials:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../getting_started.html">Getting Started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../manpages.html">Manpages</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../using_rnalib.html">Using RNAlib</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../io.html">I/O Formats</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">RNAlib API Reference</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../../api.html">Concepts and Algorithms</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../../eval.html">Free Energy Evaluation</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../grammar.html">The RNA Folding Grammar</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../landscape.html">The RNA Secondary Structure Landscape</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../mfe.html">Minimum Free Energy (MFE) Algorithms</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../pf_fold.html">Partition Function and Equilibrium Properties</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../subopt_and_representatives.html">Suboptimals and Representative Structures</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../cofold.html">RNA-RNA Interaction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../class_fold.html">Classified Dynamic Programming Variants</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../inverse.html">Inverse Folding (Design)</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../probing_data.html">Experimental Structure Probing Data</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../ligand_binding.html">Ligands Binding to RNA Structures</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../paired_modules.html">Structure Modules and Pseudoknots</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../modified_bases.html">Post-transcriptional Base Modifications</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../../utils.html">Utilities</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../alphabet.html">Utilities to deal with Nucleotide Alphabets</a></li>
<li class="toctree-l3"><a class="reference internal" href="../strings.html">(Nucleic Acid Sequence) String Utilitites</a></li>
<li class="toctree-l3"><a class="reference internal" href="../structures.html">Secondary Structure Utilities</a></li>
<li class="toctree-l3"><a class="reference internal" href="../msa.html">Multiple Sequence Alignment Utilities</a></li>
<li class="toctree-l3"><a class="reference internal" href="../io.html">Files and I/O</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="../plots.html">Plotting</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="layouts.html">Layouts and Coordinates</a></li>
<li class="toctree-l4"><a class="reference internal" href="annotation.html">Annotation</a></li>
<li class="toctree-l4"><a class="reference internal" href="probabilities.html">Pair Probability Plots</a></li>
<li class="toctree-l4 current"><a class="current reference internal" href="alignments.html#">Alignment Plots</a></li>
<li class="toctree-l4"><a class="reference internal" href="deprecated.html">Deprecated Interface for Plotting Utilities</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../search.html">Search Algorithms</a></li>
<li class="toctree-l3"><a class="reference internal" href="../combinatorics.html">Combinatorics Algorithms</a></li>
<li class="toctree-l3"><a class="reference internal" href="../data_structures.html">(Abstract) Data Structures</a></li>
<li class="toctree-l3"><a class="reference internal" href="../messages.html">Messages</a></li>
<li class="toctree-l3"><a class="reference internal" href="../logs.html">Log Messages</a></li>
<li class="toctree-l3"><a class="reference internal" href="../units.html">Unit Conversion</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../../vrna_3.0.html">RNAlib API v3.0</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../callbacks.html">Callback Functions</a></li>
<li class="toctree-l2"><a class="reference internal" href="../../deprecated.html">Deprecated List</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../wrappers.html">SWIG Wrappers</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../api_python.html">Python API</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Miscellaneous</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../news.html">What's new</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../changelog.html">Changelog</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../bibliography.html">Bibliography</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../citing.html">Citation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../license.html">License</a></li>
</ul>


    
        <p class="caption">
            <span class="caption-text">
            
                Links
            
            </span>
        </p>
        <ul>
            
                
                    <li class="toctree-l1"><a href="https://github.com/ViennaRNA/ViennaRNA"><i class="fa fa-github fa-fw"></i> Source code</a></li>
                
            
                
                    <li class="toctree-l1"><a href="https://github.com/ViennaRNA/ViennaRNA/issues"><i class="fa fa-bug fa-fw"></i> Issue tracker</a></li>
                
            
        </ul>
    


        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">ViennaRNA</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../api.html">Concepts and Algorithms</a></li>
          <li class="breadcrumb-item"><a href="../../utils.html">Utilities</a></li>
          <li class="breadcrumb-item"><a href="../plots.html">Plotting</a></li>
      <li class="breadcrumb-item active">Alignment Plots</li>

  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="alignment-plots">
<h1>Alignment Plots<a class="headerlink" href="alignments.html#alignment-plots" title="Link to this heading"></a></h1>
<p>Functions to generate Alignment plots with annotated consensus structure.</p>
<dl>
<dt class="sig sig-object cpp">
<span class="target" id="group__alignment__plots"></span></dt>
<dd><div class="breathe-sectiondef docutils container">
<p class="breathe-sectiondef-title rubric" id="breathe-section-title-functions">Functions</p>
<dl class="c function">
<dt class="sig sig-object c" id="c.vrna_file_PS_aln">
<span class="target" id="group__alignment__plots_1ga3bb7409f5b73c2273f9f76cf8b754034"></span><span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="sig-name descname"><span class="n"><span class="pre">vrna_file_PS_aln</span></span></span><span class="sig-paren">(</span><span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">filename</span></span>, <span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">seqs</span></span>, <span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">names</span></span>, <span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">structure</span></span>, <span class="kt"><span class="pre">unsigned</span></span><span class="w"> </span><span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="n"><span class="pre">columns</span></span><span class="sig-paren">)</span><a class="headerlink" href="alignments.html#c.vrna_file_PS_aln" title="Link to this definition"></a><br /></dt>
<dd><em>#include &lt;ViennaRNA/plotting/alignments.h&gt;</em><p>Create an annotated PostScript alignment plot. </p>
<p><dl>
<dt class="sig sig-object cpp">
<a class="reference internal" href="../../wrappers.html#wrappers_1_wrappers000161"><span class="std std-ref"><span class="pre">SWIG</span> <span class="pre">Wrapper</span> <span class="pre">Notes:</span></span></a></dt>
<dd><p>This function is available as overloaded function <code class="docutils literal notranslate"><span class="pre">file_PS_aln()</span></code> with three additional parameters <code class="docutils literal notranslate"><span class="pre">start</span></code>, <code class="docutils literal notranslate"><span class="pre">end</span></code>, and <code class="docutils literal notranslate"><span class="pre">offset</span></code> before the <code class="docutils literal notranslate"><span class="pre">columns</span></code> argument. Thus, it resembles the <code class="docutils literal notranslate"><a class="reference internal" href="alignments.html#group__alignment__plots_1ga2b132dddeb2044e52dc39cf0ad8afaee"><span class="std std-ref"><span class="pre">vrna_file_PS_aln_slice()</span></span></a></code> function. The last four arguments may be omitted, indicating the default of <code class="docutils literal notranslate"><span class="pre">start</span></code> = <em>0</em>, <code class="docutils literal notranslate"><span class="pre">end</span></code> = <em>0</em>, <code class="docutils literal notranslate"><span class="pre">offset</span></code> = <em>0</em>, and <code class="docutils literal notranslate"><span class="pre">columns</span></code> = <em>60</em>. See, e.g.  <span><a class="reference internal" href="../../api_python.html#RNA.file_PS_aln" title="RNA.file_PS_aln"><code class="xref py py-func docutils literal notranslate"><span class="pre">RNA.file_PS_aln()</span></code></a> in the <a class="reference internal" href="../../api_python.html"><span class="doc">Python API</span></a></span>. </p>
</dd></dl>

<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p><a class="reference internal" href="alignments.html#group__alignment__plots_1ga2b132dddeb2044e52dc39cf0ad8afaee"><span class="std std-ref">vrna_file_PS_aln_slice()</span></a></p>
</div>
</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>filename</strong> – The output file name </p></li>
<li><p><strong>seqs</strong> – The aligned sequences </p></li>
<li><p><strong>names</strong> – The names of the sequences </p></li>
<li><p><strong>structure</strong> – The consensus structure in dot-bracket notation </p></li>
<li><p><strong>columns</strong> – The number of columns before the alignment is wrapped as a new block (a value of 0 indicates no wrapping)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="c function">
<dt class="sig sig-object c" id="c.vrna_file_PS_aln_slice">
<span class="target" id="group__alignment__plots_1ga2b132dddeb2044e52dc39cf0ad8afaee"></span><span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="sig-name descname"><span class="n"><span class="pre">vrna_file_PS_aln_slice</span></span></span><span class="sig-paren">(</span><span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">filename</span></span>, <span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">seqs</span></span>, <span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">names</span></span>, <span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">structure</span></span>, <span class="kt"><span class="pre">unsigned</span></span><span class="w"> </span><span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="n"><span class="pre">start</span></span>, <span class="kt"><span class="pre">unsigned</span></span><span class="w"> </span><span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="n"><span class="pre">end</span></span>, <span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="n"><span class="pre">offset</span></span>, <span class="kt"><span class="pre">unsigned</span></span><span class="w"> </span><span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="n"><span class="pre">columns</span></span><span class="sig-paren">)</span><a class="headerlink" href="alignments.html#c.vrna_file_PS_aln_slice" title="Link to this definition"></a><br /></dt>
<dd><em>#include &lt;ViennaRNA/plotting/alignments.h&gt;</em><p>Create an annotated PostScript alignment plot. </p>
<p>Similar to <a class="reference internal" href="alignments.html#group__alignment__plots_1ga3bb7409f5b73c2273f9f76cf8b754034"><span class="std std-ref">vrna_file_PS_aln()</span></a> but allows the user to print a particular slice of the alignment by specifying a <code class="docutils literal notranslate"><span class="pre">start</span></code> and <code class="docutils literal notranslate"><span class="pre">end</span></code> position. The additional <code class="docutils literal notranslate"><span class="pre">offset</span></code> parameter allows for adjusting the alignment position ruler value.</p>
<p><dl>
<dt class="sig sig-object cpp">
<a class="reference internal" href="../../wrappers.html#wrappers_1_wrappers000162"><span class="std std-ref"><span class="pre">SWIG</span> <span class="pre">Wrapper</span> <span class="pre">Notes:</span></span></a></dt>
<dd><p>This function is available as overloaded function <code class="docutils literal notranslate"><span class="pre">file_PS_aln()</span></code> where the last four parameter may be omitted, indicating <code class="docutils literal notranslate"><span class="pre">start</span></code> = <em>0</em>, <code class="docutils literal notranslate"><span class="pre">end</span></code> = <em>0</em>, <code class="docutils literal notranslate"><span class="pre">offset</span></code> = <em>0</em>, and <code class="docutils literal notranslate"><span class="pre">columns</span></code> = <em>60</em>. See, e.g.  <span><a class="reference internal" href="../../api_python.html#RNA.file_PS_aln" title="RNA.file_PS_aln"><code class="xref py py-func docutils literal notranslate"><span class="pre">RNA.file_PS_aln()</span></code></a> in the <a class="reference internal" href="../../api_python.html"><span class="doc">Python API</span></a></span>. </p>
</dd></dl>

<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p><a class="reference internal" href="alignments.html#group__alignment__plots_1ga2b132dddeb2044e52dc39cf0ad8afaee"><span class="std std-ref">vrna_file_PS_aln_slice()</span></a></p>
</div>
</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>filename</strong> – The output file name </p></li>
<li><p><strong>seqs</strong> – The aligned sequences </p></li>
<li><p><strong>names</strong> – The names of the sequences </p></li>
<li><p><strong>structure</strong> – The consensus structure in dot-bracket notation </p></li>
<li><p><strong>start</strong> – The start of the alignment slice (a value of 0 indicates the first position of the alignment, i.e. no slicing at 5’ side) </p></li>
<li><p><strong>end</strong> – The end of the alignment slice (a value of 0 indicates the last position of the alignment, i.e. no slicing at 3’ side) </p></li>
<li><p><strong>offset</strong> – The alignment coordinate offset for the position ruler. </p></li>
<li><p><strong>columns</strong> – The number of columns before the alignment is wrapped as a new block (a value of 0 indicates no wrapping)</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="c function">
<dt class="sig sig-object c" id="c.vrna_file_PS_aln_opt">
<span class="target" id="group__alignment__plots_1gacad3969b2155a95809e1d492f8a95385"></span><span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="sig-name descname"><span class="n"><span class="pre">vrna_file_PS_aln_opt</span></span></span><span class="sig-paren">(</span><span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">filename</span></span>, <span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">seqs</span></span>, <span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">names</span></span>, <span class="k"><span class="pre">const</span></span><span class="w"> </span><span class="kt"><span class="pre">char</span></span><span class="w"> </span><span class="p"><span class="pre">*</span></span><span class="n"><span class="pre">structure</span></span>, <a class="reference internal" href="alignments.html#c.vrna_aln_opt_t" title="vrna_aln_opt_t"><span class="n"><span class="pre">vrna_aln_opt_t</span></span></a><span class="w"> </span><span class="n"><span class="pre">options</span></span><span class="sig-paren">)</span><a class="headerlink" href="alignments.html#c.vrna_file_PS_aln_opt" title="Link to this definition"></a><br /></dt>
<dd><em>#include &lt;ViennaRNA/plotting/alignments.h&gt;</em></dd></dl>

</div>
<dl class="c struct">
<dt class="sig sig-object c" id="c.vrna_aln_opt_t">
<span class="target" id="structvrna__aln__opt__t"></span><span class="k"><span class="pre">struct</span></span><span class="w"> </span><span class="sig-name descname"><span class="n"><span class="pre">vrna_aln_opt_t</span></span></span><a class="headerlink" href="alignments.html#c.vrna_aln_opt_t" title="Link to this definition"></a><br /></dt>
<dd><div class="breathe-sectiondef docutils container">
<p class="breathe-sectiondef-title rubric" id="breathe-section-title-public-members">Public Members</p>
<dl class="c var">
<dt class="sig sig-object c" id="c.vrna_aln_opt_t.start">
<span class="target" id="structvrna__aln__opt__t_1aaa82682e95e7cf29a5d752f85e38910b"></span><span class="kt"><span class="pre">unsigned</span></span><span class="w"> </span><span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="sig-name descname"><span class="n"><span class="pre">start</span></span></span><a class="headerlink" href="alignments.html#c.vrna_aln_opt_t.start" title="Link to this definition"></a><br /></dt>
<dd></dd></dl>

<dl class="c var">
<dt class="sig sig-object c" id="c.vrna_aln_opt_t.end">
<span class="target" id="structvrna__aln__opt__t_1a117ad2b78ecd7077a4902efb148cf297"></span><span class="kt"><span class="pre">unsigned</span></span><span class="w"> </span><span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="sig-name descname"><span class="n"><span class="pre">end</span></span></span><a class="headerlink" href="alignments.html#c.vrna_aln_opt_t.end" title="Link to this definition"></a><br /></dt>
<dd></dd></dl>

<dl class="c var">
<dt class="sig sig-object c" id="c.vrna_aln_opt_t.offset">
<span class="target" id="structvrna__aln__opt__t_1a68aa2cf08e6f4a703cb72ebb74781162"></span><span class="kt"><span class="pre">unsigned</span></span><span class="w"> </span><span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="sig-name descname"><span class="n"><span class="pre">offset</span></span></span><a class="headerlink" href="alignments.html#c.vrna_aln_opt_t.offset" title="Link to this definition"></a><br /></dt>
<dd></dd></dl>

<dl class="c var">
<dt class="sig sig-object c" id="c.vrna_aln_opt_t.columns">
<span class="target" id="structvrna__aln__opt__t_1a2ad9552903b95c1006618a397cfdad5f"></span><span class="kt"><span class="pre">unsigned</span></span><span class="w"> </span><span class="kt"><span class="pre">int</span></span><span class="w"> </span><span class="sig-name descname"><span class="n"><span class="pre">columns</span></span></span><a class="headerlink" href="alignments.html#c.vrna_aln_opt_t.columns" title="Link to this definition"></a><br /></dt>
<dd></dd></dl>

<dl class="c var">
<dt class="sig sig-object c" id="c.vrna_aln_opt_t.color_threshold">
<span class="target" id="structvrna__aln__opt__t_1afc0bec6a11f03e86ddc7bc6db28f3354"></span><span class="kt"><span class="pre">double</span></span><span class="w"> </span><span class="sig-name descname"><span class="n"><span class="pre">color_threshold</span></span></span><a class="headerlink" href="alignments.html#c.vrna_aln_opt_t.color_threshold" title="Link to this definition"></a><br /></dt>
<dd></dd></dl>

<dl class="c var">
<dt class="sig sig-object c" id="c.vrna_aln_opt_t.color_min_sat">
<span class="target" id="structvrna__aln__opt__t_1aeab6f09301ed4321cbe7f44591f6d756"></span><span class="kt"><span class="pre">double</span></span><span class="w"> </span><span class="sig-name descname"><span class="n"><span class="pre">color_min_sat</span></span></span><a class="headerlink" href="alignments.html#c.vrna_aln_opt_t.color_min_sat" title="Link to this definition"></a><br /></dt>
<dd></dd></dl>

</div>
</dd></dl>

</dd></dl>

</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="probabilities.html" class="btn btn-neutral float-left" title="Pair Probability Plots" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="deprecated.html" class="btn btn-neutral float-right" title="Deprecated Interface for Plotting Utilities" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 1994 - 2024, Ronny Lorenz, Ivo L. Hofacker, et al..</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>